build-tools/mono-runtimes/mono-runtimes.targets

```diff
@@ -72,6 +72,70 @@
         WorkingDirectory="$(MonoSourceFullPath)"
     />
   </Target>
+
+  <Target Name="_PrepareLlvmItems">
+    <ItemGroup>
+      <_LlvmSourceFile Include="$(LlvmSourceFullPath)\lib\**\*.cpp" />
+      <_LlvmHeaderFile Include="$(LlvmSourceFullPath)\lib\**\*.h" />
+      <_LlvmMakefileConfig Include="@(_LlvmRuntime->'$(_LlvmOutputDirTop)\%(BuildDir)\Makefile.config')" />
+      <_LlvmConfigureStamp Include="@(_LlvmRuntime->'$(_LlvmOutputDirTop)\%(BuildDir)\.stamp-configure')" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <_LlvmArchive Include="$(_LlvmOutputDirTop)\%(_LlvmRuntime.BuildDir)\Release\lib\*.a" />
+
+      <_LlvmSourceBinary Include="@(_LlvmRuntime->'$(_LlvmOutputDirTop)\%(BuildDir)\Release\bin\opt%(ExeSuffix)')" Condition=" '%(_LlvmRuntime.InstallBinaries)' == 'true' " />
+      <_LlvmTargetBinary Include="@(_LlvmRuntime->'$(OutputPath)\bin\opt%(ExeSuffix)')" Condition=" '%(_LlvmRuntime.InstallBinaries)' == 'true' "/>
+
+      <_LlvmSourceBinary Include="@(_LlvmRuntime->'$(_LlvmOutputDirTop)\%(BuildDir)\Release\bin\llc%(ExeSuffix)')" Condition=" '%(_LlvmRuntime.InstallBinaries)' == 'true' " />
+      <_LlvmTargetBinary Include="@(_LlvmRuntime->'$(OutputPath)\bin\llc%(ExeSuffix)')" Condition=" '%(_LlvmRuntime.InstallBinaries)' == 'true' " />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_ConfigureLlvm"
+          DependsOnTargets="_PrepareLlvmItems"
+          Inputs="$(LlvmSourceFullPath)\Makefile.config.in;$(LlvmSourceFullPath)\configure;@(_LlvmSourceFile);@(_LlvmHeaderFile)"
```

Comment:
Scenario: You've built "the world", then you update `external/llvm` and rebuild from topdir.

Desired behavior: the targets detect that `external/llvm` has been updated, and rebuilds LLVM and all downlevel artifacts.

Current behavior: Untested, but I assume this isn't the case, because I believe none of these `Inputs` will be outdated, unless *by chance* Makefile.config.in is changed...which probably doesn't happen all that often.

This is why e.g. the [`_SetAutogenShTimeToLastCommitTimestamp`](https://github.com/xamarin/xamarin-android/blob/a16673d/build-tools/mono-runtimes/mono-runtimes.targets#L69-L74) target (and other similar targets) sets the timestamp of a "well known file" to the timestamp of the most recent git commit. This allows for sanely re-configuring and re-building the dependency if there are more recent commits.

I would suggest that `_PrepareLlvmItems` do something similar to `_SetAutogenShTimeToLastCommitTimestamp`: pick some "well known" file (`Makefile.config.in`?), and set its timestamp, ensuring that when `external/llvm` changes, things get rebuilt.